### PR TITLE
[xwindow] part 3/n

### PR DESCRIPTION
**Pull request by Neumann-A**
adds 
- xcb-render-util
- libxkbfile
- libxkbcommon

### CODE REVIEW

**Review by LilyWangLL — DISMISSED** [review]

File: ports/libxkbcommon/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,53 @@
+if(NOT X_VCPKG_FORCE_VCPKG_X_LIBRARIES AND NOT VCPKG_TARGET_IS_WINDOWS)
+    message(STATUS "Utils and libraries provided by '${PORT}' should be provided by your system! Install the required packages or force vcpkg libraries by setting X_VCPKG_FORCE_VCPKG_X_LIBRARIES in your triplet!")
+    set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
+else()
+
```
Comment (Cheney-W) [inline ports/libxkbcommon/portfile.cmake:5]:
```suggestion

set(PATCHES "")
```

File: ports/libxkbcommon/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+)
+
+vcpkg_find_acquire_program(FLEX)
+get_filename_component(FLEX_DIR "${FLEX}" DIRECTORY )
+vcpkg_add_to_path(PREPEND "${FLEX_DIR}")
+
+vcpkg_find_acquire_program(BISON)
+get_filename_component(BISON_DIR "${BISON}" DIRECTORY )
+vcpkg_add_to_path(PREPEND "${BISON_DIR}")
+
```
Comment (Cheney-W) [inline ports/libxkbcommon/portfile.cmake:28]:
```suggestion

set(OPTIONS "")
```

File: ports/libxkbfile/portfile.cmake [hunk 3]
```diff
@@ -0,0 +1,40 @@
+if(NOT X_VCPKG_FORCE_VCPKG_X_LIBRARIES AND NOT VCPKG_TARGET_IS_WINDOWS)
+    message(STATUS "Utils and libraries provided by '${PORT}' should be provided by your system! Install the required packages or force vcpkg libraries by setting X_VCPKG_FORCE_VCPKG_X_LIBRARIES in your triplet!")
+    set(VCPKG_POLICY_EMPTY_PACKAGE enabled)
+else()
+
```
Comment (Cheney-W) [inline ports/libxkbfile/portfile.cmake:5]:
```suggestion

set(PATCHES "")
```

**Comment by LilyWangLL** [comment]
These 3 ports test pass on Linux when set `X_VCPKG_FORCE_VCPKG_X_LIBRARIES` in **x64-linux.cmake**.

**Review by ryihan — DISMISSED** [review]

**Review by ryihan — APPROVED** [review]

**Review by Cheney-W — APPROVED** [review]

**Review by LilyWangLL — APPROVED** [review]

**Comment by JavierMatosD** [comment]
Thank you!

**Pull request merged**